x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/Values.java

```diff
@@ -80,7 +82,8 @@ public Values replaceChildren(List<Expression> newChildren) {
 
     @Override
     public DataType dataType() {
-        return field().dataType();
+        DataType t = field().dataType();
```

Comment:
Instead of doing this for every aggregation function, how about going to `AggregateFunction` and define there something similar to `UnaryScalarFunction.dataType()`?